commons/h5bench_util.c

```diff
@@ -19,8 +19,6 @@
 #ifdef USE_ASYNC_VOL
 #include <H5VLconnector.h>
 #include <h5_async_lib.h>
-#else
-#include "async_adaptor.h"
```

Comment:
The problem was that it was not working. In the develop branch of HDF5 the new API had already defined `H5ES_WAIT_FOREVER` which is checked in the `async_adaptor.h`, and thus the sync calls were never used. Scot reported this issue, and after checking that was indeed the case. When I tried to make it use it for the sync tests, if we compile with VOL-async enabled (but used NON), as those calls were already defined in the new API (and they are not static) we cannot override them locally unless we use a shared library. Furthermore, as the adaptor is only used for the write and read benchmarks it is better to incorporate that into the code to solve all issues at once. Do you have any other suggestions around those issues?

Reply:
I'd prefer to keep the adaptor header file, maybe it's not useful for h5bench itself, but it can be useful for applications that has converted to _async APIs and need to compile with an older HDF5 version. I could put it in vol-async repo, but having it here allows for testing it with your existing CI more easily. We can add a version check for H5ES_WAIT_FOREVER in the header. What do you think?

Reply:
@houjun you mean keeping the file in the repository, even if we do not use it or keep including the header with a check? The way @sbyna and I were assuming (Suren, correct me if I misunderstood) is that the ASYNC would be enabled only for the newer versions of HDF5 where we have the API available (as it was included between the 1.12 and 1.13 releases). The VOL would not compile/work using older versions of HDF5, correctly?

Reply:
@jeanbez I'm thinking keeping the adaptor header and test it with one of your CI workflows with an older HDF5, maybe only enable it with a specific compile flag.

Reply:
@jeanbez, Tang's suggestion makes sense. Let's keep the adaptor header file and test it only with a flag.